docs/user-guide.md

```diff
@@ -90,13 +88,42 @@ be too. You might use this method like this:
 Tools could then use the `@Nullable` information to determine that the first use
 is safe but the second is not.
 
+As far as JSpecify is concerned, `String` and `@Nullable String` are _different_
+types. A variable of type `String` can reference any string. A variable
+of type `@Nullable String` can too, but it can also be null. This means that
+`String` is a _subtype_ of `@Nullable String`, in the same way that `Integer` is
+a subtype of `Number`. One way to look at this is that a subtype narrows the
+range of possibilities. A `Number` variable can be assigned from an `Integer`
+but it can also be assigned from a `Long`. Meanwhile an `Integer` variable
+can't be assigned from a `Number` (since that `Number` might be a `Long` or some
+other subtype). In the same way, a `@Nullable String` can be assigned from a
+`String` but a `String` can't be assigned from a `@Nullable String` (since that
+might be null).
+
+```java
+@NullMarked
+class Example {
+  private String nonNull;
+  private @Nullable String nullable;
+
+  void example() {
+	nullable = nonNull; // JSpecify allows this
+	nonNull = nullable; // JSpecify doesn't allow this
+  }
+}
+```
+
 ## `@NullMarked`
 
 The `@NullMarked` annotation indicates that references can't be null in its scope,
 unless their types are explicitly marked `@Nullable`. If applied to a package then
 its scope is all the code in the package. If applied to a class or interface
 then its scope is all the code in that class or interface.
 
+Outside `@NullMarked`, `@Nullable String` still means a reference that can be
```

Comment:
Should this be `@Nullable String x` to be consistent?

Reply:
I'm inclined to say no. You can have a return value of `@Nullable String` for example. I introduced the notion that the type `String` is a reference type by talking about variables, because I think that level of concreteness is good. Maybe at some point we should explicitly make the step from rvalue variables to expressions and from lvalue variables to method arguments. I'm not sure that's really necessary, though; readers can probably generalize on their own.

Reply:
But all your previous comments about `@Nullable String x` also apply to return values. What's different here? (and by removing the semicolon, who's to say that the very next character after `x` isn't `(`?)